controllers/operator/configmap.go

```diff
@@ -268,6 +268,16 @@ func (r *AuthenticationReconciler) handleConfigMap(instance *operatorv1alpha1.Au
 					currentConfigMap.Data["PREFERRED_LOGIN"] = newConfigMap.Data["PREFERRED_LOGIN"]
 					cmUpdateRequired = true
 				}
+				if _, keyExists := currentConfigMap.Data["AUDIT_URL"]; !keyExists {
+					reqLogger.Info("Updating an existing Configmap", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", currentConfigMap.Name)
+					currentConfigMap.Data["AUDIT_URL"] = newConfigMap.Data["AUDIT_URL"]
+					cmUpdateRequired = true
+				}
+				if _, keyExists := currentConfigMap.Data["AUDIT_SECRET"]; !keyExists {
+					reqLogger.Info("Updating an existing Configmap", "Configmap.Namespace", currentConfigMap.Namespace, "ConfigMap.Name", currentConfigMap.Name)
+					currentConfigMap.Data["AUDIT_SECRET"] = newConfigMap.Data["AUDIT_SECRET"]
+					cmUpdateRequired = true
+				}
```

Comment:
You can get rid of these - if the kvps are in the ConfigMap, we want to keep them as-is. If we had fields on the Authentication CR or some set of defaults we wanted to set, we would do that here. But not the case for this feature.

Reply:
removing it, do we need this when we upgrade and the CM exists?